### PR TITLE
Add GH action to test Scala-steward PRs

### PR DESCRIPTION
**Pull request by kelvin-chappell**
This adds an action to test Scala-steward PRs so that we can see if more work is needed before merging.
It will tell us if the code compiles and tests pass but won't tell us about runtime binary incompatibilities or if something dangerous has been imported into the repo.
So it gives more confidence but not complete confidence.

This is done as a GH action rather than in Teamcity to avoid polluting our general CI processes and generating potentially dangerous artefacts.

### CODE REVIEW

File: .github/workflows/scala-steward.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Test Scala-steward PRs
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    # Assuming Scala-steward branches are always named 'update/...' and other branches aren't.
```
Comment (mario-galic) [inline .github/workflows/scala-steward.yml:9]: Is it possible to use the branch name instead as it has `scala-steward` in it which makes it more explicit? For example

```
scala-steward:update/play-json-2.9.2
```

image

Comment (kelvin-chappell) [reply .github/workflows/scala-steward.yml:9]: [Line 17](https://github.com/guardian/support-service-lambdas/pull/858/files#diff-5f70f29f856428700109959debc5d1f950fa48d279caee7f9f0962a3caa63958R17) is to give me the name of the source branch to check that it's working but it doesn't seem to include the name of the fork so this is the best I can do I think.

Available variables are [here](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions).

Comment (mario-galic) [reply .github/workflows/scala-steward.yml:9]: Does something like this work:

```
if: ${{ github.event.pull_request.head.repo.full_name == 'scala-steward/support-service-lambdas' }}
```

Comment (kelvin-chappell) [reply .github/workflows/scala-steward.yml:9]: I thought I'd merge this and see what happens.  Then tighten it up when we get a dump of the extra environment info to find out what's available.  I'm not sure if `repo` refers to source or target.  The easiest way to find out is going to be to wait for another PR to arrive.

**Pull request merged**